File: quark/_datawire_fs_impl.js
(function () {
    "use strict";

    var os = require('os');
    var fs = require('fs');

    var quark = require('quark').quark;

    var _datawire_fs = (function () {
        function _datawire_fs() {}

        _datawire_fs.userHomeDir = function () {
            return os.homedir();
        };

        _datawire_fs.fileContents = function (path) {
            try {
                return fs.readFileSync(path, 'utf-8');
            }
            catch (e) {
                var runtime = quark.concurrent.Context.runtime();
                runtime.fail("failure reading " + path + ": " + e)
            }
        };

        return _datawire_fs;
    })();

    module.exports = _datawire_fs;
})();

File: quark/_datawire_fs_impl.py
import os

import quark

__all__ = """_datawire_fs""".split(' ')

class _datawire_fs (object):
    @classmethod
    def userHomeDir(klass):
        return os.path.expanduser('~')

    @classmethod
    def fileContents(klass, path):
        try:
            inputFile = open(path, "r")
            return inputFile.read()
        except Exception as e:
            runtime = quark.concurrent.Context.runtime()
            runtime.fail("failure reading %s: %s" % (path, e))
